addons/binding/org.openhab.binding.feed/src/main/java/org/openhab/binding/feed/handler/FeedHandler.java

```diff
@@ -0,0 +1,290 @@
+ */
+package org.openhab.binding.feed.handler;
+
+import static org.openhab.binding.feed.FeedBindingConstants.*;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.feed.FeedBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.rometools.fetcher.FeedFetcher;
+import com.rometools.fetcher.FetcherException;
+import com.rometools.fetcher.impl.FeedFetcherCache;
+import com.rometools.fetcher.impl.HashMapFeedInfoCache;
+import com.rometools.fetcher.impl.HttpURLFeedFetcher;
+import com.rometools.rome.feed.synd.SyndEntry;
+import com.rometools.rome.feed.synd.SyndFeed;
+import com.rometools.rome.io.FeedException;
+import com.rometools.rome.io.SyndFeedOutput;
+
+/**
+ *
+ * The {@link FeedHandler } is responsible for handling commands, which are
+ * sent to one of the channels and for the regular updates of the feed data.
+ *
+ * @author Svilen Valkanov - Initial contribution
+ */
+public class FeedHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(FeedHandler.class);
+
+    private String urlString;
+    private BigDecimal refreshTime;
+    /**
+     * All Feeds are converted to this format. Supported formats are {@link #SUPPORTED_FEED_FORMATS}
+     */
+    private String outputFeedFormat;
+    private BigDecimal numberOfEntriesStored;
+
+    /**
+     * FeedFetcher is used to fetch data from feed. It supports conditional GET Requests.
+     **/
+    private FeedFetcher feedFetcher;
+    private ScheduledFuture<?> refreshTask;
+    private SyndFeed currentFeedState;
+
+    public FeedHandler(Thing thing) {
+        super(thing);
+        FeedFetcherCache feedInfoCache = HashMapFeedInfoCache.getInstance();
+        feedFetcher = new HttpURLFeedFetcher(feedInfoCache);
+        currentFeedState = null;
+    }
+
+    @Override
+    public void initialize() {
+        checkConfiguration();
+
+        startAutomaticRefresh();
+    }
+
+    /**
+     * This method checks if the provided configuration is valid.
+     * When invalid parameter is found, default value is assigned.
+     */
+    private void checkConfiguration() {
+        logger.debug("Start reading Feed Thing configuration.");
+        Configuration configuration = editConfiguration();
```

Comment:
You must not edit the configuration - if you define Things in a *.things file, their configuration is read-only and cannot be changed.
Instead, use instance variables to store the values you work with (i.e. defaults if no config is set).